<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QPlainTextEdit" name="InputBox">
          <property name="plainText">
           <string>int a;
int b;

int program(int a, int b, int c)
{
	int i;
	int j;
	i = 0;
	if (a &gt; (b + c))
	{
		j = a + (b * c + 1);
	}
	else
	{
		j = a;
	}
	while (i &lt;= 100)
	{
		i = j * 2;
		j = j * 2;
	}
	return i;
}

int demo(int a)
{
	a = a + 2;
	return a * 2;
}

int main(void)
{
	int a[2][2];
	a[0][0] = 3;
	a[0][1] = a[0][0] + 1;
	a[1][0] = a[0][0] + a[0][1];
	a[1][1] = program(a[0][0], a[0][1], demo(a[1][0]));
	return a[1][1];
}
</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="compiler_button">
          <property name="text">
           <string>编译</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QPushButton" name="json_button">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>语法树结构（Json）</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="dot_button">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>语法树结构（dot）</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="midcode_button">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>中间代码</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QPushButton" name="IBlock_button">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>基本块</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="optimizer_block_button">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>优化后中间代码</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QPushButton" name="object_Iblocks_button">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>待用活跃信息</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="object_code_button">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>目标代码</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>输出</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="outBox"/>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3"/>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>compiler_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>Lex_Parser()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>203</x>
     <y>472</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>json_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>json_button()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>73</x>
     <y>499</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dot_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>dot_button()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>203</x>
     <y>499</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>midcode_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>midcode_button()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>332</x>
     <y>499</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>IBlock_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>IBlock_button()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>106</x>
     <y>527</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>optimizer_block_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>optimizer_block_button()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>300</x>
     <y>527</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>object_Iblocks_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>object_Iblocks_button()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>106</x>
     <y>555</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>object_code_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>object_code_button()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>300</x>
     <y>555</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>Lex_Parser()</slot>
  <slot>json_button()</slot>
  <slot>dot_button()</slot>
  <slot>midcode_button()</slot>
  <slot>IBlock_button()</slot>
  <slot>optimizer_block_button()</slot>
  <slot>object_Iblocks_button()</slot>
  <slot>object_code_button()</slot>
 </slots>
</ui>
